server/server_test.go

```diff
@@ -6310,6 +6315,80 @@ func TestServer_UpdateConfig(t *testing.T) {
 	db := NewInfluxDB(func(q string) *iclient.Response {
 		return &iclient.Response{}
 	})
+	defMap := map[string]interface{}{
+		"default":                     false,
+		"disable-subscriptions":       false,
+		"enabled":                     true,
+		"excluded-subscriptions":      map[string]interface{}{"_kapacitor": []interface{}{"autogen"}},
+		"http-port":                   float64(0),
+		"insecure-skip-verify":        false,
+		"kapacitor-hostname":          "",
+		"name":                        "default",
+		"password":                    true,
+		"ssl-ca":                      "",
+		"ssl-cert":                    "",
+		"ssl-key":                     "",
+		"startup-timeout":             "1h0m0s",
+		"subscription-protocol":       "http",
+		"subscription-mode":           "cluster",
+		"subscription-path":           "",
+		"subscriptions":               nil,
+		"subscriptions-sync-interval": "1m0s",
+		"timeout":                     "0s",
+		"udp-bind":                    "",
+		"udp-buffer":                  float64(1e3),
+		"udp-read-buffer":             float64(0),
+		"urls":                        []interface{}{db.URL()},
+		"username":                    "bob",
+		"compression":                 "gzip",
+	}
+
+	deepCopyMapWithReplace := func(m map[string]interface{}) func(replacements ...map[string]interface{}) map[string]interface{} {
+		var a interface{} = "" // we have to do this to prevent gob.Register from panicing
+		gob.Register([]string{})
+		gob.Register(map[string]interface{}{})
+		gob.Register([]interface{}{})
+		gob.Register(a)
+		return func(replacements ...map[string]interface{}) map[string]interface{} {
+			// we can't just use json here because json(foo) doesn't always equal decodedJson(foo)
+			var buf bytes.Buffer        // Stand-in for a network connection
+			enc := gob.NewEncoder(&buf) // Will write to network.
+			dec := gob.NewDecoder(&buf) // Will read from network.
+			if err := enc.Encode(m); err != nil {
+				t.Fatal(err)
+			}
+			mCopy := map[string]interface{}{}
+
+			if err := dec.Decode(&mCopy); err != nil {
+				t.Fatal(err)
+			}
+			for i := range replacements {
+				for k, v := range replacements[i] {
+					v := v
+					if err := enc.Encode(v); err != nil {
+						t.Fatal(err)
+					}
+					vCopy := reflect.Indirect(reflect.New(reflect.TypeOf(v)))
+					if err := dec.DecodeValue(vCopy); err != nil {
+						t.Fatal(err)
+					}
+					mCopy[k] = vCopy.Interface()
+				}
+			}
+			return mCopy
+		}
+	}
+	if !cmp.Equal(deepCopyMapWithReplace(defMap)(), defMap) {
+		t.Fatalf("deepCopyMapWithReplace is broken expected %v, got %v", defMap, deepCopyMapWithReplace(defMap)())
+	}
+	{ // new block to keep vars clean
```

Comment:
IMO the extra indentation from this makes it less readable, but not the hill I'm going to die on 😄